experimental/devices/pca9548/pca9548.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9548
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/physic"
+)
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{Address: 0x70}
+
+// Opts is the pca9548 configuration.
+type Opts struct {
+	// Address pca9548 I²C Address.
+	Address int
+}
+
+// Dev is handle to a pca9548 I²C Multiplexer.
+type Dev struct {
+	// Immutable.
+	c       i2c.Bus
+	address uint16
+
+	// Mutable.
+	mu   sync.Mutex
+	port uint8
+}
+
+// Register creates a new handle to a pca9548 I²C multiplexer, and registers
+// port names with the host. These ports can then be used as any other i2c.Bus.
+// The registered port names are in the form: pca9548/mux-ADD-I where ADD is the
+// multiplexer I²C address in hex and I is the port number.
+// example: "pca9548-70-0" and "mux-70-0".
+func Register(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	d := &Dev{
+		c:       bus,
+		port:    0xFF,
+		address: uint16(opts.Address),
+	}
+	r := make([]byte, 1)
+	err := bus.Tx(uint16(opts.Address), nil, r)
+	if err != nil {
+		return nil, errors.New("could not communicated with multiplexer: " + err.Error())
+	}
+
+	for i := uint8(0); i < 8; i++ {
+		portID := strconv.FormatUint(uint64(i), 10)
+		addrStr := strconv.FormatUint(uint64(opts.Address), 16)
+		name := addrStr + "-" + portID
+		opener := newOpener(d, i)
+		if err := i2creg.Register("pca9548-"+name, []string{"mux-" + name}, int((opts.Address*10 + int(i))), opener); err != nil {
```

Comment:
In practice, this would fail if an host instantiate this driver twice. There's nothing on that hardware side to do so. Even funnier would be to daisy chain one.

Reply:
tbh, I'm not sure on the difference between the alias and the name, should the alias be provided by the user eg:
```golang
 func (d *Dev) Register(port int, alias string) error
```
might make it a bit more friendly as they would then know what the port is called or does this make more sense?

```golang
 func (d *Dev) Open(port int) (i2c.Bus, error)
```